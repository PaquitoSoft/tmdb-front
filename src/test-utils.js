import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppProvider } from './components/app-context/app-context';

const userIdMock = 'hskfjahshjdjskla';

const imagesConfigMock = {
	"baseUrl":"https://image.tmdb.org/t/p/",
	"backdropSizes":["w300","w780","w1280","original"],
	"logoSizes":["w45","w92","w154","w185","w300","w500","original"],
	"posterSizes":["w92","w154","w185","w342","w500","w780","original"],
	"profileSizes":["w45","w185","h632","original"],
	"stillSizes":["w92","w185","w300","original"]
};

const renderers = {
	'shallow': shallow,
	'mount': mount
};

function AppContextMock({ children, apiClientMock }) {
	return (
		<AppProvider 
			apiClient={apiClientMock} 
			userId={userIdMock} 
			imagesConfig={imagesConfigMock}
		>
			{children}
		</AppProvider>
	);
}

export function render(
	component, 
	options = { renderType: 'shallow' }
) {
	return renderers[options.renderType](component);
}

export function validateSnapshot(
	component, 
	options = { renderType: 'shallow' }
) {
	const wrapper = render(component, options);
	expect(wrapper).toMatchSnapshot();
	return wrapper;
}

export function renderInAppContext(
	component, 
	options = {}
) {
	const _options = {
		renderType: 'mount',
		apiClientQueryMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		apiClientMutationMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		...options
	};

	const apiClientMock = {
		sendQuery: _options.apiClientQueryMock,
		sendMutation: _options.apiClientMutationMock
	};
	
	const wrapper = render(
		(
			<AppContextMock apiClientMock={apiClientMock}>
				{component}
			</AppContextMock>
		),
		_options
	);

	return wrapper.find(AppProvider).childAt(0);
}

export const clone = object => JSON.parse(JSON.stringify(object));
