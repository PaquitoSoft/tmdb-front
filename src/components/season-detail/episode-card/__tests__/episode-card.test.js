import React from 'react';
import { validateSnapshot, render } from '../../../../test-utils';

import EpisodeCard from '../episode-card';

describe('EpisodeCard', () => {

	const episodeMockData = {
		"id": 1974881,
		"name": "Betrayal",
		"episodeNumber": 8,
		"imagePath": "/qtvvzZz2ACzg971VifhmqY4vONb.jpg",
		"airDate": "2019-12-22",
		"voteAverage": 0,
		"overview": "As the Magisterium closes in, Lyra learns more about Asrielâ€™s rebellion. But her assistance to him comes at great personal cost."
	};

	it('Should render episode card', () => {
		validateSnapshot(<EpisodeCard episode={episodeMockData} />);
	});

	it('Should show episode overview when clicking the overview link', () => {
		const wrapper = render(<EpisodeCard episode={episodeMockData} />);
		
		expect(wrapper.exists('.episode-card__overview-panel')).toBe(false);
		wrapper.find('.episode-card__overview-link').simulate('click');
		expect(wrapper.exists('.episode-card__overview-panel')).toBe(true);
	});
});
