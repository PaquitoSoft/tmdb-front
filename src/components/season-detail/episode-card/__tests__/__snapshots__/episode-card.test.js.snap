// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCard Should render episode card 1`] = `
<div
  className="episode-card"
>
  <MediaImage
    alt="Betrayal"
    className="episode-card__media"
    path="/qtvvzZz2ACzg971VifhmqY4vONb.jpg"
    ratio={1.77}
    sizes="30vw"
    type="still"
  />
  <div
    className="episode-card__info"
  >
    <div
      className="episode-card__row"
    >
      <div
        className="episode-card__title"
      >
        <span
          className="episode-card__number"
        >
          8
          .Â 
        </span>
        <span
          className="episode-card__name "
        >
          Betrayal
        </span>
      </div>
    </div>
    <div
      className="episode-card__row"
    >
      <span
        className="episode-card__date"
      >
        (
        Sun Dec 22 2019
        )
      </span>
      <span
        className="episode-card__overview-link"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        See overview
      </span>
    </div>
  </div>
</div>
`;
